src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -454,6 +470,93 @@ public AgentSettings LoadSettings()
             }
         }
 
+#if OS_WINDOWS
+        private void ConfigureAutoLogonIfNeeded(CommandSettings command)
+        {
+            Trace.Info(nameof(ConfigureAutoLogonIfNeeded));
+
+            bool enableAutoLogon = command.GetEnableAutoLogon();
+            if(!enableAutoLogon)
+            {
+                Trace.Info("AutoLogon will not be enabled as per user's input.");
+                return;
+            }
+
+            try
+            {
+                AssertAdminAccess();
+                
+                var iConfigManager = HostContext.GetService<IInteractiveSessionConfigurationManager>();
+                iConfigManager.Configure(command);
+
+                if(iConfigManager.RestartNeeded())
+                {
+                    Trace.Info("AutoLogon is configured for a different user than the current user. Machine needs a restart.");
+                    _term.WriteLine(StringUtil.Loc("RestartMessage"));
+                    var shallRestart = command.GetRestartPermission();
+                    if(shallRestart)
+                    {
+                        Trace.Info("Restarting the machine in 5 seconds");
+                        _term.WriteLine(StringUtil.Loc("RestartIn5SecMessage"));
+                        Process.Start("shutdown.exe", "-r -t 5");
```

Comment:
And WhichUtil (or expand %windir%\System32\shutdown.exe and IOUtil to assert the file exists).

What else does the agent need to do before the agent process terminates? Is 5 seconds enough?

Shutdown accepts a comment. It seems like it would be good to add an information message there (localized).

I noticed that shutdown indicates it uses "/" for args/switches. And that you used "-". We support back to Win 7, so as long as "-" works back to Win 7 then it's OK. If not, I wonder if "/" is better supported since that is what is indicated in the command help.